x/stakeibc/keeper/msg_server_delete_trade_route.go

```diff
@@ -2,33 +2,28 @@ package keeper
 
 import (
 	"context"
-	"fmt"
+
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
 
 	"github.com/Stride-Labs/stride/v16/x/stakeibc/types"
 
 	errorsmod "cosmossdk.io/errors"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
-func (k msgServer) DeleteTradeRoute(goCtx context.Context, msg *types.MsgDeleteTradeRoute) (*types.MsgDeleteTradeRouteResponse, error) {
+func (ms msgServer) DeleteTradeRoute(goCtx context.Context, msg *types.MsgDeleteTradeRoute) (*types.MsgDeleteTradeRouteResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
-	k.Logger(ctx).Info(fmt.Sprintf("delete trade route: %s", msg.String()))
-
-	// get our addresses, make sure they're valid
-	_, err := sdk.AccAddressFromBech32(msg.Creator)
-	if err != nil {
-		return nil, errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "creator address is invalid: %s. err: %s", msg.Creator, err.Error())
+	if ms.authority != msg.Authority {
+		return nil, errorsmod.Wrapf(govtypes.ErrInvalidSigner, "invalid authority; expected %s, got %s", ms.authority, msg.Authority)
 	}
 
-	routes := k.GetAllTradeRoutes(ctx)
-	for _, route := range routes {
-		if route.TargetDenomOnHostZone == msg.HostDenom && route.RewardDenomOnRewardZone == msg.RewardDenom {
-			k.RemoveTradeRoute(ctx, route.RewardDenomOnHostZone, route.TargetDenomOnHostZone)
-		}
-		// if no matching trade route was found for the given host-denom and reward-denom... do nothing
+	_, found := ms.Keeper.GetTradeRoute(ctx, msg.RewardDenom, msg.HostDenom)
```

Comment:
Why have this check that it exists?  If delete succeeds silently when it can't find the route and when it actually removes the route then it becomes idempotent which is often nice.

Reply:
I think it's just a slightly nicer UX in the event that you were to fat finger either of the denom's.

Since this is gov prop, you can imagine a scenario where we have a typo but don't find out for 3 days later until the prop passes and we notice it was never deleted. Whereas, if we add a check here, simulating this tx for the gov prop will fail if the input is invalid

Reply:
hmm ok.  Yes that is a good argument I'm sold